client/src/App.jsx

```diff
@@ -39,11 +40,13 @@ const router = createBrowserRouter([
 
 const App = () => (
   <React.StrictMode>
-    <DataProvider>
-      <ModalProvider>
-        <RouterProvider router={router} />
-      </ModalProvider>
-    </DataProvider>
+    <PageProvider>
+      <DataProvider>
```

Comment:
PageProvider wrapped around DataProvider so DataProvider has access to Page context